uart_abstraction/BUILD.bazel

```diff
@@ -0,0 +1,11 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+cc_library(
+    name = "fake",
+    srcs = [
+        "fake/HardwareSerial.cpp",
+    ],
+    hdrs = ["fake/HardwareSerial.hpp"],
```

Comment:
Will this header eventually be shared by multiple implementations?

Reply:
Good point. I've moved things around etc.